### PR TITLE
docs: refactor style guide example 03-06

### PR DESCRIPTION
**Pull request by danielsogl**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Style guide example 03-06 uses the old Angular Http module

## What is the new behavior?
Updated the example to use thew new HttpClient 

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by danielsogl** [comment]
cc @brandonroberts

**Review by brandonroberts — APPROVED** [review]

**Comment by vicb** [comment]
Failing the CI with the error:

```
ERROR in src/03-06/app/heroes/shared/hero.service.avoid.ts(24,32): error TS2339: Property 'data' does not exist on type 'Hero'.
src/03-06/app/heroes/shared/hero.service.avoid.ts(29,32): error TS2339: Property 'data' does not exist on type 'Hero[]'.
```

**Comment by danielsogl** [comment]
@vicb should be fixed now

File: aio/content/examples/styleguide/src/03-06/app/heroes/shared/hero.service.avoid.ts [hunk 1]
```diff
@@ -16,18 +16,18 @@ export class HeroService {
     private exceptionService: ExceptionService,
     private spinnerService: SpinnerService,
     private toastService: ToastService,
-    private http: Http
-  ) { }
+    private http: HttpClient
+  ) {}
```
Comment (gkalpak) [inline aio/content/examples/styleguide/src/03-06/app/heroes/shared/hero.service.avoid.ts]: Keep the space (for consistency).

File: aio/content/examples/styleguide/src/03-06/app/heroes/shared/hero.service.avoid.ts [hunk 2]
```diff
@@ -16,18 +16,18 @@ export class HeroService {
...
   getHeroes() {
-    return this.http.get(`api/heroes`).pipe(
-      map(response => response.json().data as Hero[]));
+    return this.http
+      .get<Hero[]>(`api/heroes`)
+      .pipe(map((response: any) => response.data as Hero[]));
   }
-
 }
-
```
Comment (gkalpak) [inline aio/content/examples/styleguide/src/03-06/app/heroes/shared/hero.service.avoid.ts]: Other than the imports, the rest of this file should be identical to [hero.service.ts][1].

[1]: https://github.com/danielsogl/angular/blob/2047abe91413aa13f10b549efa989a6852b81ecd/aio/content/examples/styleguide/src/03-06/app/heroes/shared/hero.service.ts

File: aio/content/examples/styleguide/src/03-06/app/heroes/shared/hero.service.ts [hunk 3]
```diff
@@ -16,17 +17,17 @@ export class HeroService {
...
-      map(response => response.json() as Hero));
+    return this.http.get<Hero>(`api/heroes/${id}`).pipe(
+      map(response => response as Hero));
   }
 
   getHeroes() {
-    return this.http.get(`api/heroes`).pipe(
-      map(response => response.json() as Hero[]));
+    return this.http.get<Hero[]>(`api/heroes`).pipe(
+      map(response => response as Hero[]));
```
Comment (gkalpak) [inline aio/content/examples/styleguide/src/03-06/app/heroes/shared/hero.service.ts]: The `pipe(map(...))` is no longer necessary (here and above).

**Review by gkalpak — CHANGES_REQUESTED** [review]

**Comment by danielsogl** [comment]
@gkalpak Fixed

File: aio/content/examples/styleguide/src/03-06/app/heroes/shared/hero.service.avoid.ts [hunk 4]
```diff
@@ -16,18 +15,14 @@ export class HeroService {
...
-      map(response => response.json().data as Hero));
+    return this.http.get<Hero>(`api/heroes/${id}`);
   }
 
   getHeroes() {
-    return this.http.get(`api/heroes`).pipe(
-      map(response => response.json().data as Hero[]));
+    return this.http.get<Hero[]>(`api/heroes`);
   }
-
```
Comment (gkalpak) [inline aio/content/examples/styleguide/src/03-06/app/heroes/shared/hero.service.avoid.ts]: This line should stay 😇

**Review by gkalpak — APPROVED** [review]
One last nit. LGTM otherwise (assuming CI is green) 👍

File: aio/content/examples/styleguide/src/03-06/app/heroes/shared/hero.service.avoid.ts (hunk 4 above)
Comment (danielsogl) [reply aio/content/examples/styleguide/src/03-06/app/heroes/shared/hero.service.avoid.ts]: When I open the file there is a free line after the last curly bracket and in GitHub Web too.

Comment (danielsogl) [reply aio/content/examples/styleguide/src/03-06/app/heroes/shared/hero.service.avoid.ts]: Ahhh damit 😅

**Comment by vicb** [comment]
Thanks for the update, merging again.

**Pull request closed without merging**